### ISSUE TITLE
Error trying to have multiple projects that use project_options

### ISSUE THREAD

**Issue by mCodingLLC** [aminya/project_options#149]
There's a decent chance this is pure user error and I don't understand how to compose project_options properly, but I'm getting configure-time errors when trying to have multiple projects using project_options, specifically:

```
[cmake] CMake Error at build/_deps/_project_options-src/src/Index.cmake:125 (add_library):
[cmake]   add_library cannot create target "project_options" because another target
[cmake]   with the same name already exists.  The existing target is an interface
[cmake]   library created in source directory
[cmake]   "/home/mcoding/Projects/project_options_multiple_include_mwe".  See
[cmake]   documentation for policy CMP0002 for more details.
```

The project structure is like this:

```
project/
    apps/
        CMakeLists.txt
        main.cpp
    CMakeLists.txt
    external/mylib/
        CMakeLists.txt
        include/mylib
            mylib.hpp
```

Both `project` and the submodule `mylib` use `project_options`, which explains why cmake thinks `project_options` is being multiply defined, because both `project` and `mylib` do indeed define their own set of project options. Is this a situation where `project_options` should be defining target names that depend on the project name like `${PROJECT_NAME}_project_options` instead of a global name `project_options`? Or is there some way to have `project_options` scoped to a namespace like `mylib::project_options`? Or should only the top-level project be defining project options (using an `if TARGET` guard before including it) with subprojects modifying the options using `dynamic_project_options`?

Any guidance here would be appreciated.

Here are the relevant list files for the minimal working example:

Top level `CMakeLists.txt`:
```
cmake_minimum_required(VERSION 3.16)

include(FetchContent)
FetchContent_Declare(_project_options URL https://github.com/aminya/project_options/archive/refs/tags/v0.24.1.zip)
FetchContent_MakeAvailable(_project_options)
include(${_project_options_SOURCE_DIR}/Index.cmake)
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)

project(myproject LANGUAGES CXX C)

dynamic_project_options()

add_subdirectory(external/mylib)

add_subdirectory(apps)
```

`apps/CMakeLists.txt`
```
add_executable(main main.cpp)
target_link_libraries(main PRIVATE project_options project_warnings)

target_link_system_libraries(
  main
  PRIVATE mylib
)
```

`external/mylib/CMakeLists.txt`
```
cmake_minimum_required(VERSION 3.16)

include(FetchContent)
FetchContent_Declare(_project_options URL https://github.com/aminya/project_options/archive/refs/tags/v0.24.1.zip)
FetchContent_MakeAvailable(_project_options)
include(${_project_options_SOURCE_DIR}/Index.cmake)
include(${_project_options_SOURCE_DIR}/src/DynamicProjectOptions.cmake)

project(mylib LANGUAGES CXX C)

dynamic_project_options()

add_library(mylib INTERFACE)
target_link_libraries(mylib INTERFACE project_options project_warnings)

set(INCLUDE_DIR "include") 
target_include_directories(mylib INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${INCLUDE_DIR}>"
                                          "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
```

**Comment by ClausKlein** [comment]
One solution would be, the every `project_options-` `cmake cache variable` and `interface library` must be prefixed with the `${PROJECT_NAME}.`

@mCodingLLC by the way, your project names should be different!

**Comment by aminya** [comment]
We also need to check if project_options is a target, and if it is we realize it is a multi-project.

**Comment by aminya** [comment]
I made a pull request to fix this issue. #150

**Comment by mCodingLLC** [comment]
Wow very fast fix, much appreciated!

Since I'm the first to bring this up, has this not been an issue before because you don't generally use/recommend submodules within a repo as the place to store dependencies? If so, where do you keep your dependencies? Install them and always use find_package?

**Comment by aminya** [comment]
Released in v0.25.0


> Since I'm the first to bring this up, has this not been an issue before because you don't generally use/recommend submodules within a repo as the place to store dependencies? If so, where do you keep your dependencies? Install them and always use find_package?

I recommend using vcpkg. See 
https://github.com/aminya/cpp_vcpkg_project
